plugins/hypervisors/vmware/src/main/java/com/cloud/storage/resource/VmwareStorageProcessor.java

```diff
@@ -2822,7 +2822,10 @@ private ManagedObjectReference getVmfsDatastore(VmwareContext context, VmwareHyp
             throw new Exception("A relevant SCSI disk could not be located to use to create a datastore.");
         }
 
-        morDs = firstHostDatastoreSystemMO.createVmfsDatastore(datastoreName, hostScsiDisk);
+        morDs = firstHostDatastoreSystemMO.findDatastoreByName(datastoreName);
```

Comment:
Will it be better to use `HypervisorHostHelper.findDatastoreWithBackwardsCompatibility` instead? @harikrishna-patnala @nvazquez @weizhouapache

Reply:
@shwstppr 
this uses same code as line 2810
we could change both lines if you agree this is better.

Reply:
@weizhouapache probably safe to use `firstHostDatastoreSystemMO.findDatastoreByName` then. I've not played with the managed storages myself.
From what I understand, HypervisorHostHelper.findDatastoreWithBackwardsCompatibility tries to find the datastore using custom field(cloud.uuid) set for the datastore. Not sure if we even do that for managed datastores.

Reply:
agree with what you said. @shwstppr 
too bad we do not have a managed storage to verify it